packages/pluggableWidgets/badge-native/src/Badge.tsx

```diff
@@ -1,51 +1,40 @@
 import { flattenStyles } from "@native-mobile-resources/util-widgets";
-import { Component, createElement } from "react";
+import { createElement, ReactElement, useCallback } from "react";
 import { Platform, Text, TouchableNativeFeedback, TouchableOpacity, View } from "react-native";
 
 import { BadgeProps } from "../typings/BadgeProps";
 import { BadgeStyle, defaultBadgeStyle } from "./ui/Styles";
-import { executeAction } from "@widgets-resources/piw-utils";
+import { executeAction, isAvailable } from "@widgets-resources/piw-utils";
 
 export type Props = BadgeProps<BadgeStyle>;
 
-export class Badge extends Component<Props> {
-    private readonly onClickHandler = this.onClick.bind(this);
-    private readonly styles = flattenStyles(defaultBadgeStyle, this.props.style);
-
-    render(): JSX.Element {
-        const isAndroid = Platform.OS === "android";
-
-        return (
-            <View style={this.styles.container} testID={this.props.name}>
-                {this.props.onClick ? (
-                    isAndroid ? (
-                        <TouchableNativeFeedback
-                            background={TouchableNativeFeedback.SelectableBackgroundBorderless()}
-                            onPress={this.onClickHandler}
-                        >
-                            {this.renderText()}
-                        </TouchableNativeFeedback>
-                    ) : (
-                        <TouchableOpacity onPress={this.onClickHandler}>{this.renderText()}</TouchableOpacity>
-                    )
-                ) : (
-                    this.renderText()
-                )}
-            </View>
-        );
-    }
-
-    private renderText(): JSX.Element {
-        const value = this.props.caption.value || "";
+export function Badge(props: BadgeProps<BadgeStyle>): ReactElement {
+    const styles = flattenStyles(defaultBadgeStyle, props.style);
 
+    const renderText = useCallback(() => {
         return (
-            <Text testID={`${this.props.name}$caption`} style={this.styles.caption}>
-                {value}
+            <Text testID={`${props.name}$caption`} style={styles.caption}>
+                {isAvailable(props.caption) ? props.caption.value : ""}
             </Text>
         );
-    }
-
-    private onClick(): void {
-        executeAction(this.props.onClick);
-    }
+    }, [props.caption.value, props.caption.status, styles]);
+
+    return (
+        <View style={styles.container} testID={props.name}>
+            {props.onClick ? (
+                Platform.OS === "android" ? (
+                    <TouchableNativeFeedback
```

Comment:
Can we use `Pressable` instead of `TouchableNativeFeedback` and `TouchableOpacity` ?

Reply:
Done, while doing this i realized that ripple color was not working at all so implemented that. This means we need to do the necessary adjustments for the atlas which will be a TODO